reactor-core/src/main/java/reactor/core/publisher/MonoZip.java

```diff
@@ -333,13 +339,26 @@ public void onNext(Object t) {
 		@Override
 		public void onError(Throwable t) {
 			error = t;
-			parent.signalError(t);
+			if (parent.delayError) {
+				parent.signal();
+			}
+			else {
+				int n = parent.subscribers.length;
+				if (ZipCoordinator.DONE.getAndSet(parent, n) != n) {
+					parent.cancel(this);
+					parent.actual.onError(t);
+				}
+			}
 		}
 
 		@Override
 		public void onComplete() {
 			if (value == null) {
```

Comment:
should the `delayError` be taken into account there too? eg. if one source is empty but delay error is active, we wait for other sources to complete? Also, this doesn't take the possibility that delayed error have been encountered by other inners... once again, I'd keep the `parent.signal` method and amend the behavior in the `signal()` method itself to cover all bases (delayError with 1 erroring source and 1 empty source)